### PR TITLE
Fix some type issues detected by Psalm

### PR DESCRIPTION
**Pull request by lptn**
Few fixes in types and PHPDoc to make codebase more readable for static analysis tools (PHPStan, Psalm, IDEs, etc)

### CODE REVIEW

**Comment by szepeviktor** [comment]
@toonvandenbos This PR is very valuable. Could you take a look at it?

File: src/Value/Resolver.php
Comment (toonvandenbos) [inline src/Value/Resolver.php:20]: I'm not a big fan of these changes. Why would `$value` be a better var name than `$groups` when using named arguments? (real question)

File: src/Value/Resolver.php
Comment (toonvandenbos) [inline src/Value/Resolver.php:41]: Same here, why would `$model` be better than `$resource` for named arguments? This argument is `mixed` and should therefore not be confused with a model (it could be another Layout instance, or any other container object depending on the project).

File: src/Value/Resolver.php
Comment (szepeviktor) [reply src/Value/Resolver.php:20]: Because `value` is reserved word.
`$this->set(value: $value, model: $model, attribute: $attr);` would fail.

Comment (lptn) [reply src/Value/Resolver.php:20]: @toonvandenbos 
agree, sorry for not mentioning this. This is because the implemented [interface has this name](https://github.com/whitecube/nova-flexible-content/blob/7a5f9276e89c990f259852021f13811b8c59328a/src/Value/ResolverInterface.php#L7 ) plus I couldn't find `@no-named-arguments` tags or other docs/promises of supporting/not supporting named arguments introduced in PHP 8

Comment (lptn) [reply src/Value/Resolver.php:20]: PS: I'm happy to revert these changes if it will help to merge this PR

Comment (szepeviktor) [reply src/Value/Resolver.php:20]: And there is [uninformative-words](https://github.com/szepeviktor/uninformative-words).
But it may not spark joy!

Comment (toonvandenbos) [reply src/Value/Resolver.php:20]: Thank you for the insights. These arguments have indeed not been named consistently.

IMO the "clean" solution would be to rename [`$value` into `$groups` in `Value/ResolverInterface.php`](https://github.com/whitecube/nova-flexible-content/blob/7a5f9276e89c990f259852021f13811b8c59328a/src/Value/ResolverInterface.php#L7). This change should not break any existing implementations, plus we could avoid the uninformative `$value` variable (and more importantly that ugly `$groups = $value` statement which is honestly hurting my soul 😅).

Comment (toonvandenbos) [reply src/Value/Resolver.php:20]: @lptn you can still edit this PR. I'll wait for your changes and give it a final review before merging. Thanks!

Comment (lptn) [reply src/Value/Resolver.php:20]: @toonvandenbos 
done: renamed, also added some PHPDoc to the interface + updated a stub + extra cleanup (safe one)

Comment (szepeviktor) [reply src/Value/Resolver.php:20]: @lptn but forgot to commit

**Review by szepeviktor — APPROVED** [review]

File: src/Value/Resolver.php
Comment (lptn) [reply src/Value/Resolver.php]: @toonvandenbos 
Please review again when you have a time - the PR is ready. Thanks!

**Review by toonvandenbos — APPROVED** [review]

**Pull request merged**
